### DISCUSSION TITLE
Allow HTML in Repeater Item Label

### DISCUSSION THREAD

**Discussion by ronfuller** [filamentphp/filament#13477] [Ideas]
Would it be possible to allow the repeater item label to contain HTML. It is advantageous to provide status icons or badges in the Repeater Item labels. 

This would be a simple change to the repeater/index.blade.php file , line 154 of current release

```
  <h4
      @class([
          'text-sm font-medium text-gray-950 dark:text-white',
          'truncate' => $isItemLabelTruncated(),
      ])
  >
      {!! $itemLabel !!}
  </h4>
```
Example shown with status badge icons for Repeater Item Labels

<img width="1013" alt="image" src="https://github.com/filamentphp/filament/assets/9759221/925da1ec-38e8-472b-84b7-eb75383af332">

@danharrin I am willing to submit a PR if you are okay with this request.

**Comment by danharrin** [comment]
You can use an HtmlString object or another Htmlable object like a View to achieve this already. Implementing your suggestion would introduce an XSS vulnerability.

**Reply by ronfuller** [reply]
Perfect!
I have used HtmlString throughout the app in other places. I must have missed this one.

This code works perfectly. Thank you for the timely response.  And Filament is amazing! Thank you for building this platform.

```
 ->itemLabel(fn (array $state): HtmlString => new HtmlString($this->getSectionItemLabelString($state)))
 ```

**Reply by misteryomi** [reply]
An example implementation, for anyone who's looking to do this is:

` ->itemLabel(fn (array $state): HtmlString => new HtmlString($state['title']. ' - <a href="'.RecommendationResource::getUrl('edit', ['record' => $state['slug']]).'">Edit</a>') ?? null)
`

**Comment by abdulmannans-joistic** [comment]
@danharrin  Can we make  item label multiline ?
<img width="1415" alt="Screenshot 2024-09-11 at 2 52 48 PM" src="https://github.com/user-attachments/assets/4d82c219-3282-4c12-b5e4-d0231154ffb1">
